src/Parser/OrientationParser.js

```diff
@@ -0,0 +1,111 @@
+import * as THREE from 'three';
+
+const DEG2RAD = THREE.Math.DEG2RAD;
+
+// The transform from world to local is  RotationZ(heading).RotationX(pitch).RotationY(roll)
+// The transform from local to world is (RotationZ(heading).RotationX(pitch).RotationY(roll)).transpose()
+THREE.Quaternion.prototype.setFromRollPitchHeading = function setFromRollPitchHeading(roll = 0, pitch = 0, heading = 0) {
+    roll *= DEG2RAD;
+    pitch *= DEG2RAD;
+    heading *= DEG2RAD;
+    // return this.setFromEuler(new THREE.Euler(pitch, roll, heading , 'ZXY')).conjugate();
+    return this.setFromEuler(new THREE.Euler(-pitch, -roll, -heading, 'YXZ')); // optimized version of above
+};
+
+// From DocMicMac, the transform from local to world is:
+// RotationX(omega).RotationY(phi).RotationZ(kappa).RotationX(PI)
+// RotationX(PI) = Scale(1, -1, -1) converts between the 2 conventions for the camera local frame:
+//  X right, Y bottom, Z front : convention in webGL, threejs and computer vision
+//  X right, Y top,    Z back  : convention in photogrammetry
+THREE.Quaternion.prototype.setFromOmegaPhiKappa = function setFromOmegaPhiKappa(omega = 0, phi = 0, kappa = 0) {
+    omega *= DEG2RAD;
+    phi *= DEG2RAD;
+    kappa *= DEG2RAD;
+    this.setFromEuler(new THREE.Euler(omega, phi, kappa, 'XYZ'));
+    // this.setFromRotationMatrix(new THREE.Matrix4().makeRotationFromQuaternion(this).scale(new THREE.Vector3(1, -1, -1)));
+    this.set(this.w, this.z, -this.y, -this.x); // optimized version of above
+    return this;
+};
+
+/**
+ * @function setENUFromGeodesicNormal
+ * @param {Vector3} up - the normalized geodetic normal to the ellipsoid (given by Coordinates.geodeticNormal)
+ * @returns {Quaternion} this
+ */
+THREE.Quaternion.prototype.setENUFromGeodesicNormal = (() => {
+    const matrix = new THREE.Matrix4();
+    const elements = matrix.elements;
+    const north = new THREE.Vector3();
+    const east = new THREE.Vector3();
+    return function setENUFromGeodesicNormal(up) {
+        // this is an optimized version of matrix.lookAt(up, new THREE.Vector3(0, 0, 0), new THREE.Vector3(0, 0, 1));
```

Comment:
Could you elaborate on this optimization ?

Reply:
the (0,0,0) and (0,0,1) values just make a few operations of Matrix4.lookAt trivial
(I wrote this code :) )

Reply:
I read too fast once again ;)

Reply:
How do you handle the case for `up == Vector3(0, 0, 0)` ? Does the result (being a matrix filled with 0 if I'm correct) works afterwards ?